src/fu-tool.c

```diff
@@ -3932,9 +3935,149 @@ fu_util_reboot_cleanup(FuUtilPrivate *priv, gchar **values, GError **error)
+
+	optional_data = fu_efi_load_option_get_optional_data(entry);
+	if (optional_data != NULL) {
+		g_autofree gchar *safestr =
+		    fu_strsafe((const gchar *)g_bytes_get_data(optional_data, NULL),
+			       g_bytes_get_size(optional_data));
+		fwupd_codec_string_append(str, indent, "EntryOptionalData", safestr);
+	}
+
+	/* all entries should have a list */
+	dp_list =
+	    fu_firmware_get_image_by_gtype(FU_FIRMWARE(entry), FU_TYPE_EFI_DEVICE_PATH_LIST, error);
+	if (dp_list == NULL)
+		return FALSE;
+
+	/* FILE */
+	dp_path = FU_EFI_FILE_PATH_DEVICE_PATH(
+	    fu_firmware_get_image_by_gtype(FU_FIRMWARE(dp_list),
+					   FU_TYPE_EFI_FILE_PATH_DEVICE_PATH,
+					   NULL));
+	if (dp_path != NULL) {
+		g_autofree gchar *name = fu_efi_file_path_device_path_get_name(dp_path, error);
+		if (name == NULL)
+			return FALSE;
+		fwupd_codec_string_append(str, indent, "EntryFilePath", name);
+	}
+
+	/* HDD */
+	dp_hdd = FU_EFI_HARD_DRIVE_DEVICE_PATH(
+	    fu_firmware_get_image_by_gtype(FU_FIRMWARE(dp_list),
+					   FU_TYPE_EFI_HARD_DRIVE_DEVICE_PATH,
+					   NULL));
+	if (dp_hdd != NULL) {
+		g_autoptr(FuVolume) volume = NULL;
+		g_autofree gchar *partition_signature = fwupd_guid_to_string(
+		    fu_efi_hard_drive_device_path_get_partition_signature(dp_hdd),
+		    FWUPD_GUID_FLAG_MIXED_ENDIAN);
+
+		fwupd_codec_string_append(str, indent, "PartitionSignature", partition_signature);
+		fwupd_codec_string_append_hex(
+		    str,
+		    indent,
+		    "PartitionNumber",
+		    fu_efi_hard_drive_device_path_get_partition_number(dp_hdd));
+		fwupd_codec_string_append_hex(
+		    str,
+		    indent,
+		    "PartitionSize",
+		    fu_efi_hard_drive_device_path_get_partition_size(dp_hdd));
+		fwupd_codec_string_append_hex(
+		    str,
+		    indent,
+		    "PartitionOffset",
+		    fu_efi_hard_drive_device_path_get_partition_start(dp_hdd));
+
+		/* can we match the volume? */
+		volume =
+		    fu_context_get_esp_volume_by_hard_drive_device_path(priv->ctx, dp_hdd, error);
+		if (volume != NULL) {
+			fwupd_codec_string_append(str,
+						  indent,
+						  "MountPoint",
+						  fu_volume_get_mount_point(volume));
+		}
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_util_efivar_files(FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	g_autoptr(GPtrArray) files = NULL;
+
+	files = fu_context_get_esp_files(priv->ctx,
+					 FU_CONTEXT_ESP_FILE_FLAG_INCLUDE_FIRST_STAGE |
+					     FU_CONTEXT_ESP_FILE_FLAG_INCLUDE_SECOND_STAGE,
+					 error);
+	if (files == NULL)
+		return FALSE;
+	for (guint i = 0; i < files->len; i++) {
+		FuFirmware *firmware = g_ptr_array_index(files, i);
+		g_autofree gchar *name =
+		    g_strdup_printf("Boot%04X", (guint)fu_firmware_get_idx(firmware));
+		fu_console_print(priv->console,
+				 "%s → %s",
+				 name,
+				 fu_firmware_get_filename(firmware));
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_util_efivar_boot(FuUtilPrivate *priv, gchar **values, GError **error)
```

Comment:
This seems to be missing json support (IE `fwupdtool efivar-boot --json` is the same as without).  I think for the vision of maybe displacing efibootmgr at some point we'll want to add json to all the commands.

Reply:
Added!